### DISCUSSION TITLE
Retain counter value after reset

### DISCUSSION THREAD

**Discussion by cyberax17** [arendst/Tasmota#10854] [Q&A] [answered]
Dears, 
I have configured tasmota with a counter which counts pulses on a pin of my esp and sends the counter value via mqtt. I realised that the counter value is reset to 0 after a power cycle of the esp. 
is there any means to configure this behaviour and, eventually save the counter value so that it is not lost after a reset ? 
many thanks for your help in advance!

**Comment by sfromis** [comment, marked as answer]
It is generally saved at midnight, but there is no option to save it after every change, as this could wear down the flash chip too fast. The thing is that there is a limited number of flash write cycle available before it fails. You could implement rules to do more frequent saves, balancing your flash lifetime expectancy.

An alternate strategy is to handle it backend, where you could publish a command to the device after boot, to restore the "last seen" counter value.

**Reply by cyberax17** [reply]
Thanks , this is already a good information. However in my case the counter was reset to 0 after a power cycle and not to the last value the midnight before, what I would expect according to your description. Or am I getting something wrong ? What am I missing there ? 
I think I will also implement your proposal to preset the counter via the backend in such a case.

**Reply by sfromis** [reply]
Well, I'm not actually certain of midnight, it may "just" be when something else triggers a save. I may have gotten it mixed up with energy monitoring values, which are saved at midnight.

**Comment by ascillato** [comment]
The counter value is never saved on flash. If you want to keep it, you must use a rule and a MEM

**Reply by sfromis** [reply]
Actually, the counters are saved in flash when other settings are saved. I've seen that on devices many times powered off. But they do not "count" as settings changes, so the updates do not trigger a save.

**Reply by ascillato** [reply]
Yes, you are right. They are saved. If you trigger something that makes a save on flash, the counter's value survives the power cycle.
You can change the value of a relay, do a restart, etc.
